app/controllers/catalog_controller.rb

```diff
@@ -1140,6 +1140,18 @@ def service_dialog_from_ot_submit_save
     end
   end
 
+  def service_dialog_generator(template)
+    # TODO: the special cases for Amazon and Azure should be removed once the service class is moved to the provider gem
+    if template.kind_of? OrchestrationTemplateCfn
+      return OrchestrationTemplateCfnDialogService
+    elsif template.kind_of? OrchestrationTemplateAzure
+      return OrchestrationTemplateAzureDialogService
+    end
+
+    # TODO: if a template can be used by multiple manager types, the UI should provide a dropdown box for user to select
+    "#{template.class.eligible_manager_types.first}::#{template.class.name.demodulize}DialogService".constantize
```

Comment:
Yes. Stack options are per manager type while parameters are per template. So the options for the same template can be different depending on which manager is selected.